// shaders/dist/core/shaderRegistry-DFzzFEKX.js
import { n as componentDefinition } from "./AngularBlur-BgZWbEjj.js";
import { n as componentDefinition$1 } from "./Ascii-BwSMmD6Z.js";
import { n as componentDefinition$2 } from "./Aurora-KcUDr49P.js";
import { n as componentDefinition$3 } from "./Beam-CAnoOWgt.js";
import { n as componentDefinition$4 } from "./Blob-CuRSRXmA.js";
import { n as componentDefinition$5 } from "./Blur-DG7lJKPo.js";
import { n as componentDefinition$6 } from "./BrightnessContrast-CJTOpQCn.js";
import { n as componentDefinition$7 } from "./Bulge-Chpbxxzj.js";
import { n as componentDefinition$8 } from "./ChannelBlur-BhcCkwcO.js";
import { n as componentDefinition$9 } from "./Checkerboard-ucave4-a.js";
import { n as componentDefinition$10 } from "./ChromaFlow-DxQaUqPL.js";
import { n as componentDefinition$11 } from "./ChromaticAberration-DsMj-vBw.js";
import { n as componentDefinition$12 } from "./Circle-QgCuVo44.js";
import { n as componentDefinition$13 } from "./ConcentricSpin-Crfxti93.js";
import { n as componentDefinition$14 } from "./ConicGradient-CQAPYlQg.js";
import { n as componentDefinition$15 } from "./ContourLines-tDBvQOvb.js";
import { n as componentDefinition$16 } from "./Crescent-BkRt-XWy.js";
import { n as componentDefinition$17 } from "./Cross-kkY7feuC.js";
import { n as componentDefinition$18 } from "./CRTScreen-CTwXU-NX.js";
import { n as componentDefinition$19 } from "./Crystal-COGiLVvV.js";
import { n as componentDefinition$20 } from "./CursorRipples-D-cfjHY5.js";
import { n as componentDefinition$21 } from "./CursorTrail-CgRMXbTG.js";
import { n as componentDefinition$22 } from "./DiamondGradient-BC3ugP8a.js";
import { n as componentDefinition$23 } from "./DiffuseBlur-eyM1v3ad.js";
import { n as componentDefinition$24 } from "./Dither-CDfDotR6.js";
import { n as componentDefinition$25 } from "./DOMTexture-D-RkeyMs.js";
import { n as componentDefinition$26 } from "./DotGrid-T9YbgC3V.js";
import { n as componentDefinition$27 } from "./DropShadow-B7FzTp4H.js";
import { n as componentDefinition$28 } from "./Duotone-CqQ0NMxS.js";
import { n as componentDefinition$29 } from "./Ellipse-CXxGqTYI.js";
import { n as componentDefinition$30 } from "./Emboss-LU-6cckT.js";
import { n as componentDefinition$31 } from "./FallingLines-CArEXajJ.js";
import { n as componentDefinition$32 } from "./FilmGrain-DSJiMtEp.js";
import { n as componentDefinition$33 } from "./FloatingParticles-DpYA0_TE.js";
import { n as componentDefinition$34 } from "./Flower-MAwlZEv8.js";
import { n as componentDefinition$35 } from "./FlowField-RyqiRxyI.js";
import { n as componentDefinition$36 } from "./FlowingGradient-C0Zoyfd6.js";
import { n as componentDefinition$37 } from "./Fog-BeKcPcTq.js";
import { n as componentDefinition$38 } from "./Form3D-Bufw6tZb.js";
import { n as componentDefinition$39 } from "./Glass-Bz9IKI2q.js";
import { n as componentDefinition$40 } from "./GlassTiles-Dgh6nOaE.js";
import { n as componentDefinition$41 } from "./Glitch-BsbzLahF.js";
import { n as componentDefinition$42 } from "./Glow-CUv1yHLZ.js";
import { n as componentDefinition$43 } from "./Godrays-D1LINbut.js";
import { n as componentDefinition$44 } from "./Grayscale-Ri-695Uh.js";
import { n as componentDefinition$45 } from "./Grid-BYSsAIEv.js";
import { n as componentDefinition$46 } from "./GridDistortion-SRyevgsp.js";
import { n as componentDefinition$47 } from "./Group-CkhjTW59.js";
import { n as componentDefinition$48 } from "./Halftone-BUCDTTAU.js";
import { n as componentDefinition$49 } from "./HexGrid-BUiDrehv.js";
import { n as componentDefinition$50 } from "./HueShift-ctEX5vfo.js";
import { n as componentDefinition$51 } from "./ImageTexture-uxj4LpF7.js";
import { n as componentDefinition$52 } from "./Invert-azYrcDMi.js";
import { n as componentDefinition$53 } from "./Kaleidoscope-CFWtKisp.js";
import { n as componentDefinition$54 } from "./LensFlare-ttrXgfea.js";
import { n as componentDefinition$55 } from "./LinearBlur-By2aD8OO.js";
import { n as componentDefinition$56 } from "./LinearGradient-CM02z32l.js";
import { n as componentDefinition$57 } from "./Liquify-p-_qcb1f.js";
import { n as componentDefinition$58 } from "./Mirror-B7XmfGdw.js";
import { n as componentDefinition$59 } from "./MultiPointGradient-8zGSIM4U.js";
import { n as componentDefinition$60 } from "./Neon-CMmGJzPy.js";
import { n as componentDefinition$61 } from "./Paper-B08rxNys.js";
import { n as componentDefinition$62 } from "./Perspective-BN_OpTZq.js";
import { n as componentDefinition$63 } from "./Pixelate-dxPAFCRu.js";
import { n as componentDefinition$64 } from "./Plasma-DGCw0cd7.js";
import { n as componentDefinition$65 } from "./PolarCoordinates-CMYCFT4S.js";
import { n as componentDefinition$66 } from "./Polygon-CsCOt5YE.js";
import { n as componentDefinition$67 } from "./Posterize-Blv9pzPT.js";
import { n as componentDefinition$68 } from "./ProgressiveBlur-NbvxFfRa.js";
import { n as componentDefinition$69 } from "./RadialGradient-De0gtdh3.js";
import { n as componentDefinition$70 } from "./RectangularCoordinates-8Sa2GaDy.js";
import { n as componentDefinition$71 } from "./Ring-kHH-DIDQ.js";
import { n as componentDefinition$72 } from "./Ripples-Bf6Zwg6X.js";
import { n as componentDefinition$73 } from "./RoundedRect-B0sWh2BN.js";
import { n as componentDefinition$74 } from "./Saturation-DCFAIUvZ.js";
import { n as componentDefinition$75 } from "./Sharpness-DSpEoCSL.js";
import { n as componentDefinition$76 } from "./Shatter-ByKkCYTR.js";
import { n as componentDefinition$77 } from "./SimplexNoise-CMuU8n27.js";
import { n as componentDefinition$78 } from "./SineWave-VTNDSK_g.js";
import { n as componentDefinition$79 } from "./Smoke-9MJIAKq7.js";
import { n as componentDefinition$80 } from "./SmokeFill-qV2iRIts.js";
import { n as componentDefinition$81 } from "./Solarize-CWDgSBiv.js";
import { n as componentDefinition$82 } from "./SolidColor-DhMdM3bE.js";
import { n as componentDefinition$83 } from "./Spherize-BZxwopX8.js";
import { n as componentDefinition$84 } from "./Spiral-gpkw37l1.js";
import { n as componentDefinition$85 } from "./Star-WJAOZ5P5.js";
import { n as componentDefinition$86 } from "./Strands-DOthZNHt.js";
import { n as componentDefinition$87 } from "./Stretch--GV6hbru.js";
import { n as componentDefinition$88 } from "./Stripes-CFCXw1_N.js";
import { n as componentDefinition$89 } from "./StudioBackground-DJr-cxOL.js";
import { n as componentDefinition$90 } from "./Swirl-B_KxdaX2.js";
import { n as componentDefinition$91 } from "./TiltShift-BSAyR0Me.js";
import { n as componentDefinition$92 } from "./Tint-DdvnpzEJ.js";
import { n as componentDefinition$93 } from "./Trapezoid-Co0sIA73.js";
import { n as componentDefinition$94 } from "./Tritone-Ckb6hTkI.js";
import { n as componentDefinition$95 } from "./Truchet-CkFSkniR.js";
import { n as componentDefinition$96 } from "./Twirl-v8pLqUTN.js";
import { n as componentDefinition$97 } from "./Vesica-CxP7UD0a.js";
import { n as componentDefinition$98 } from "./VHS-mKd1QBBj.js";
import { n as componentDefinition$99 } from "./Vibrance-WR5yQv_9.js";
import { n as componentDefinition$100 } from "./VideoTexture-BMSdfLf_.js";
import { n as componentDefinition$101 } from "./Vignette-hAIKTi35.js";
import { n as componentDefinition$102 } from "./Voronoi-DOGmwUSK.js";
import { n as componentDefinition$103 } from "./WaveDistortion-BGdgj3Iz.js";
import { n as componentDefinition$104 } from "./Weave-CIP_EUPa.js";
import { n as componentDefinition$105 } from "./WebcamTexture-Ds9IrtRn.js";
import { n as componentDefinition$106 } from "./ZoomBlur-BuIVpHGG.js";
var createShaderRegistry = () => {
	const registry = {};
	const shaderDefinitions = {
		AngularBlur: componentDefinition,
		Ascii: componentDefinition$1,
		Aurora: componentDefinition$2,
		Beam: componentDefinition$3,
		Blob: componentDefinition$4,
		Blur: componentDefinition$5,
		BrightnessContrast: componentDefinition$6,
		Bulge: componentDefinition$7,
		ChannelBlur: componentDefinition$8,
		Checkerboard: componentDefinition$9,
		ChromaFlow: componentDefinition$10,
		ChromaticAberration: componentDefinition$11,
		Circle: componentDefinition$12,
		ConcentricSpin: componentDefinition$13,
		ConicGradient: componentDefinition$14,
		ContourLines: componentDefinition$15,
		Crescent: componentDefinition$16,
		Cross: componentDefinition$17,
		CRTScreen: componentDefinition$18,
		Crystal: componentDefinition$19,
		CursorRipples: componentDefinition$20,
		CursorTrail: componentDefinition$21,
		DiamondGradient: componentDefinition$22,
		DiffuseBlur: componentDefinition$23,
		Dither: componentDefinition$24,
		DOMTexture: componentDefinition$25,
		DotGrid: componentDefinition$26,
		DropShadow: componentDefinition$27,
		Duotone: componentDefinition$28,
		Ellipse: componentDefinition$29,
		Emboss: componentDefinition$30,
		FallingLines: componentDefinition$31,
		FilmGrain: componentDefinition$32,
		FloatingParticles: componentDefinition$33,
		Flower: componentDefinition$34,
		FlowField: componentDefinition$35,
		FlowingGradient: componentDefinition$36,
		Fog: componentDefinition$37,
		Form3D: componentDefinition$38,
		Glass: componentDefinition$39,
		GlassTiles: componentDefinition$40,
		Glitch: componentDefinition$41,
		Glow: componentDefinition$42,
		Godrays: componentDefinition$43,
		Grayscale: componentDefinition$44,
		Grid: componentDefinition$45,
		GridDistortion: componentDefinition$46,
		Group: componentDefinition$47,
		Halftone: componentDefinition$48,
		HexGrid: componentDefinition$49,
		HueShift: componentDefinition$50,
		ImageTexture: componentDefinition$51,
		Invert: componentDefinition$52,
		Kaleidoscope: componentDefinition$53,
		LensFlare: componentDefinition$54,
		LinearBlur: componentDefinition$55,
		LinearGradient: componentDefinition$56,
		Liquify: componentDefinition$57,
		Mirror: componentDefinition$58,
		MultiPointGradient: componentDefinition$59,
		Neon: componentDefinition$60,
		Paper: componentDefinition$61,
		Perspective: componentDefinition$62,
		Pixelate: componentDefinition$63,
		Plasma: componentDefinition$64,
		PolarCoordinates: componentDefinition$65,
		Polygon: componentDefinition$66,
		Posterize: componentDefinition$67,
		ProgressiveBlur: componentDefinition$68,
		RadialGradient: componentDefinition$69,
		RectangularCoordinates: componentDefinition$70,
		Ring: componentDefinition$71,
		Ripples: componentDefinition$72,
		RoundedRect: componentDefinition$73,
		Saturation: componentDefinition$74,
		Sharpness: componentDefinition$75,
		Shatter: componentDefinition$76,
		SimplexNoise: componentDefinition$77,
		SineWave: componentDefinition$78,
		Smoke: componentDefinition$79,
		SmokeFill: componentDefinition$80,
		Solarize: componentDefinition$81,
		SolidColor: componentDefinition$82,
		Spherize: componentDefinition$83,
		Spiral: componentDefinition$84,
		Star: componentDefinition$85,
		Strands: componentDefinition$86,
		Stretch: componentDefinition$87,
		Stripes: componentDefinition$88,
		StudioBackground: componentDefinition$89,
		Swirl: componentDefinition$90,
		TiltShift: componentDefinition$91,
		Tint: componentDefinition$92,
		Trapezoid: componentDefinition$93,
		Tritone: componentDefinition$94,
		Truchet: componentDefinition$95,
		Twirl: componentDefinition$96,
		Vesica: componentDefinition$97,
		VHS: componentDefinition$98,
		Vibrance: componentDefinition$99,
		VideoTexture: componentDefinition$100,
		Vignette: componentDefinition$101,
		Voronoi: componentDefinition$102,
		WaveDistortion: componentDefinition$103,
		Weave: componentDefinition$104,
		WebcamTexture: componentDefinition$105,
		ZoomBlur: componentDefinition$106
	};
	Object.entries(shaderDefinitions).forEach(([fileName, def]) => {
		const propsMetadata = {};
		if (def.props) Object.entries(def.props).forEach(([propName, propConfig]) => {
			if (propConfig.ui) propsMetadata[propName] = {
				ui: propConfig.ui,
				default: propConfig.default,
				description: propConfig.description
			};
		});
		registry[def.name] = {
			name: def.name,
			fileName,
			category: def.category || "Uncategorized",
			definition: def,
			propsMetadata
		};
	});
	return registry;
};
const shaderRegistry = createShaderRegistry();
function getAllShaders() {
	return Object.values(shaderRegistry);
}
function getShaderByName(name) {
	return shaderRegistry[name];
}
function getShadersByCategory(category) {
	return Object.values(shaderRegistry).filter((shader) => shader.category === category);
}
function getShaderCategories() {
	const categories = new Set(Object.values(shaderRegistry).map((shader) => shader.category));
	return Array.from(categories).sort();
}
export { shaderRegistry as a, getShadersByCategory as i, getShaderByName as n, getShaderCategories as r, getAllShaders as t };
