// shaders/dist/partner/index.js
const DEFAULT_BASE_URL = "https://shaders.com";
const CACHE_TTL_MS = 5 * 60 * 1e3;
class PartnerClient {
  constructor(options) {
    this.cache = /* @__PURE__ */ new Map();
    if (!options.apiKey) throw new Error("PartnerClient requires an apiKey");
    this.apiKey = options.apiKey;
    this.baseUrl = (options.baseUrl ?? DEFAULT_BASE_URL).replace(/\/$/, "");
  }
  getCached(key) {
    const entry = this.cache.get(key);
    if (!entry) return null;
    if (Date.now() > entry.expiresAt) {
      this.cache.delete(key);
      return null;
    }
    return entry.data;
  }
  setCached(key, data) {
    this.cache.set(key, { data, expiresAt: Date.now() + CACHE_TTL_MS });
  }
  async listPresets(options = {}) {
    const page = Math.max(1, options.page ?? 1);
    const limit = Math.min(100, Math.max(1, options.limit ?? 20));
    const cacheKey = `presets:p${page}:l${limit}`;
    const cached = this.getCached(cacheKey);
    if (cached) return cached;
    const url = new URL(`${this.baseUrl}/api/partner/v1/presets`);
    url.searchParams.set("page", String(page));
    url.searchParams.set("limit", String(limit));
    const response = await fetch(url.toString(), {
      headers: { Authorization: `Bearer ${this.apiKey}` }
    });
    if (!response.ok) {
      let message = response.statusText;
      try {
        const body = await response.json();
        if (body.message) message = body.message;
      } catch {
      }
      throw new Error(`Partner API error ${response.status}: ${message}`);
    }
    const data = await response.json();
    this.setCached(cacheKey, data);
    return data;
  }
}
export {
  PartnerClient
};
