// shaders/dist/js/createPreview.js
import { createShader } from "./createShader.js";
import { decodePreviewDefinition } from "./decode.js";
var DEFAULT_API_BASE = "https://shaders.com";
var WATERMARK_TEXT = "Unlock your Shaders Pro license";
var WATERMARK_LINK = "https://shaders.com/dashboard?pricing=true";
async function createPreview(canvas, options) {
	if (!options.shader && !options.presetId || options.shader && options.presetId) throw new Error("Exactly one of shader (preview token) or presetId must be provided");
	const apiBase = options.apiBaseUrl || DEFAULT_API_BASE;
	const url = options.shader ? `${apiBase}/api/preview/shader/${encodeURIComponent(options.shader)}` : `${apiBase}/api/preview/preset/${encodeURIComponent(options.presetId)}`;
	const response = await fetch(url);
	if (!response.ok) throw new Error(`Failed to fetch preview: ${response.status} ${response.statusText}`);
	const instance = await createShader(canvas, decodePreviewDefinition(await response.text()), {
		isPreview: true,
		enablePerformanceTracking: true
	});
	const { overlay, savedPosition, parentEl } = createWatermarkOverlay(canvas);
	const originalDestroy = instance.destroy.bind(instance);
	instance.destroy = () => {
		if (parentEl && savedPosition !== void 0) parentEl.style.position = savedPosition;
		overlay.remove();
		originalDestroy();
	};
	return instance;
}
function createWatermarkOverlay(canvas) {
	const parent = canvas.parentElement;
	let savedPosition;
	if (parent) {
		if (getComputedStyle(parent).position === "static") {
			savedPosition = parent.style.position;
			parent.style.position = "relative";
		}
	}
	const link = document.createElement("a");
	link.href = WATERMARK_LINK;
	link.target = "_blank";
	link.rel = "noopener noreferrer";
	link.textContent = WATERMARK_TEXT;
	Object.assign(link.style, {
		position: "absolute",
		bottom: "8px",
		right: "12px",
		fontSize: "11px",
		fontFamily: "system-ui, -apple-system, sans-serif",
		color: "rgba(255, 255, 255, 0.5)",
		textDecoration: "none",
		zIndex: "10",
		pointerEvents: "auto",
		transition: "color 0.2s ease"
	});
	link.addEventListener("mouseenter", () => {
		link.style.color = "rgba(255, 255, 255, 0.8)";
	});
	link.addEventListener("mouseleave", () => {
		link.style.color = "rgba(255, 255, 255, 0.5)";
	});
	if (parent) parent.appendChild(link);
	return {
		overlay: link,
		savedPosition,
		parentEl: parent
	};
}
export { createPreview };
