// shaders/dist/solid/engine/Preview.js
import { template, insert, memo, createComponent, effect, className, style, setAttribute, Dynamic, mergeProps } from "solid-js/web";
import { createSignal, createEffect, createMemo, For } from "solid-js";
import ShaderComponent$1G from "../components/AngularBlur.js";
import ShaderComponent$1F from "../components/Ascii.js";
import ShaderComponent$1E from "../components/Aurora.js";
import ShaderComponent$1D from "../components/Beam.js";
import ShaderComponent$1C from "../components/Blob.js";
import ShaderComponent$1B from "../components/Blur.js";
import ShaderComponent$1A from "../components/BrightnessContrast.js";
import ShaderComponent$1z from "../components/Bulge.js";
import ShaderComponent$1y from "../components/CRTScreen.js";
import ShaderComponent$1x from "../components/ChannelBlur.js";
import ShaderComponent$1w from "../components/Checkerboard.js";
import ShaderComponent$1v from "../components/ChromaFlow.js";
import ShaderComponent$1u from "../components/ChromaticAberration.js";
import ShaderComponent$1t from "../components/Circle.js";
import ShaderComponent$1s from "../components/ConcentricSpin.js";
import ShaderComponent$1r from "../components/ConicGradient.js";
import ShaderComponent$1q from "../components/ContourLines.js";
import ShaderComponent$1p from "../components/Crescent.js";
import ShaderComponent$1o from "../components/Cross.js";
import ShaderComponent$1n from "../components/Crystal.js";
import ShaderComponent$1m from "../components/CursorRipples.js";
import ShaderComponent$1l from "../components/CursorTrail.js";
import ShaderComponent$1k from "../components/DOMTexture.js";
import ShaderComponent$1j from "../components/DiamondGradient.js";
import ShaderComponent$1i from "../components/DiffuseBlur.js";
import ShaderComponent$1h from "../components/Dither.js";
import ShaderComponent$1g from "../components/DotGrid.js";
import ShaderComponent$1f from "../components/DropShadow.js";
import ShaderComponent$1e from "../components/Duotone.js";
import ShaderComponent$1d from "../components/Ellipse.js";
import ShaderComponent$1c from "../components/Emboss.js";
import ShaderComponent$1b from "../components/FallingLines.js";
import ShaderComponent$1a from "../components/FilmGrain.js";
import ShaderComponent$19 from "../components/FloatingParticles.js";
import ShaderComponent$18 from "../components/FlowField.js";
import ShaderComponent$17 from "../components/Flower.js";
import ShaderComponent$16 from "../components/FlowingGradient.js";
import ShaderComponent$15 from "../components/Fog.js";
import ShaderComponent$14 from "../components/Form3D.js";
import ShaderComponent$13 from "../components/Glass.js";
import ShaderComponent$12 from "../components/GlassTiles.js";
import ShaderComponent$11 from "../components/Glitch.js";
import ShaderComponent$10 from "../components/Glow.js";
import ShaderComponent$$ from "../components/Godrays.js";
import ShaderComponent$_ from "../components/Grayscale.js";
import ShaderComponent$Z from "../components/Grid.js";
import ShaderComponent$Y from "../components/GridDistortion.js";
import ShaderComponent$X from "../components/Group.js";
import ShaderComponent$W from "../components/Halftone.js";
import ShaderComponent$V from "../components/HexGrid.js";
import ShaderComponent$U from "../components/HueShift.js";
import ShaderComponent$T from "../components/ImageTexture.js";
import ShaderComponent$S from "../components/Invert.js";
import ShaderComponent$R from "../components/Kaleidoscope.js";
import ShaderComponent$Q from "../components/LensFlare.js";
import ShaderComponent$P from "../components/LinearBlur.js";
import ShaderComponent$O from "../components/LinearGradient.js";
import ShaderComponent$N from "../components/Liquify.js";
import ShaderComponent$M from "../components/Mirror.js";
import ShaderComponent$L from "../components/MultiPointGradient.js";
import ShaderComponent$K from "../components/Neon.js";
import ShaderComponent$J from "../components/Paper.js";
import ShaderComponent$I from "../components/Perspective.js";
import ShaderComponent$H from "../components/Pixelate.js";
import ShaderComponent$G from "../components/Plasma.js";
import ShaderComponent$F from "../components/PolarCoordinates.js";
import ShaderComponent$E from "../components/Polygon.js";
import ShaderComponent$D from "../components/Posterize.js";
import ShaderComponent$C from "../components/ProgressiveBlur.js";
import ShaderComponent$B from "../components/RadialGradient.js";
import ShaderComponent$A from "../components/RectangularCoordinates.js";
import ShaderComponent$z from "../components/Ring.js";
import ShaderComponent$y from "../components/Ripples.js";
import ShaderComponent$x from "../components/RoundedRect.js";
import ShaderComponent$w from "../components/Saturation.js";
import ShaderComponent$v from "../components/Sharpness.js";
import ShaderComponent$u from "../components/Shatter.js";
import ShaderComponent$t from "../components/SimplexNoise.js";
import ShaderComponent$s from "../components/SineWave.js";
import ShaderComponent$r from "../components/Smoke.js";
import ShaderComponent$q from "../components/SmokeFill.js";
import ShaderComponent$p from "../components/Solarize.js";
import ShaderComponent$o from "../components/SolidColor.js";
import ShaderComponent$n from "../components/Spherize.js";
import ShaderComponent$m from "../components/Spiral.js";
import ShaderComponent$l from "../components/Star.js";
import ShaderComponent$k from "../components/Strands.js";
import ShaderComponent$j from "../components/Stretch.js";
import ShaderComponent$i from "../components/Stripes.js";
import ShaderComponent$h from "../components/StudioBackground.js";
import ShaderComponent$g from "../components/Swirl.js";
import ShaderComponent$f from "../components/TiltShift.js";
import ShaderComponent$e from "../components/Tint.js";
import ShaderComponent$d from "../components/Trapezoid.js";
import ShaderComponent$c from "../components/Tritone.js";
import ShaderComponent$b from "../components/Truchet.js";
import ShaderComponent$a from "../components/Twirl.js";
import ShaderComponent$9 from "../components/VHS.js";
import ShaderComponent$8 from "../components/Vesica.js";
import ShaderComponent$7 from "../components/Vibrance.js";
import ShaderComponent$6 from "../components/VideoTexture.js";
import ShaderComponent$5 from "../components/Vignette.js";
import ShaderComponent$4 from "../components/Voronoi.js";
import ShaderComponent$3 from "../components/WaveDistortion.js";
import ShaderComponent$2 from "../components/Weave.js";
import ShaderComponent$1 from "../components/WebcamTexture.js";
import ShaderComponent from "../components/ZoomBlur.js";
import Shader from "./Shader.js";
var _tmpl$ = /* @__PURE__ */ template(`<div style=position:relative;width:100%;height:100%><a target=_blank rel="noopener noreferrer"style="position:absolute;bottom:8px;right:12px;font-size:11px;font-family:-apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, sans-serif;color:rgba(255, 255, 255, 0.5);text-decoration:none;pointer-events:auto;z-index:10;text-shadow:0 1px 3px rgba(0, 0, 0, 0.4);transition:color 0.2s ease">`);
const componentMap = {
  AngularBlur: ShaderComponent$1G,
  Ascii: ShaderComponent$1F,
  Aurora: ShaderComponent$1E,
  Beam: ShaderComponent$1D,
  Blob: ShaderComponent$1C,
  Blur: ShaderComponent$1B,
  BrightnessContrast: ShaderComponent$1A,
  Bulge: ShaderComponent$1z,
  CRTScreen: ShaderComponent$1y,
  ChannelBlur: ShaderComponent$1x,
  Checkerboard: ShaderComponent$1w,
  ChromaFlow: ShaderComponent$1v,
  ChromaticAberration: ShaderComponent$1u,
  Circle: ShaderComponent$1t,
  ConcentricSpin: ShaderComponent$1s,
  ConicGradient: ShaderComponent$1r,
  ContourLines: ShaderComponent$1q,
  Crescent: ShaderComponent$1p,
  Cross: ShaderComponent$1o,
  Crystal: ShaderComponent$1n,
  CursorRipples: ShaderComponent$1m,
  CursorTrail: ShaderComponent$1l,
  DOMTexture: ShaderComponent$1k,
  DiamondGradient: ShaderComponent$1j,
  DiffuseBlur: ShaderComponent$1i,
  Dither: ShaderComponent$1h,
  DotGrid: ShaderComponent$1g,
  DropShadow: ShaderComponent$1f,
  Duotone: ShaderComponent$1e,
  Ellipse: ShaderComponent$1d,
  Emboss: ShaderComponent$1c,
  FallingLines: ShaderComponent$1b,
  FilmGrain: ShaderComponent$1a,
  FloatingParticles: ShaderComponent$19,
  FlowField: ShaderComponent$18,
  Flower: ShaderComponent$17,
  FlowingGradient: ShaderComponent$16,
  Fog: ShaderComponent$15,
  Form3D: ShaderComponent$14,
  Glass: ShaderComponent$13,
  GlassTiles: ShaderComponent$12,
  Glitch: ShaderComponent$11,
  Glow: ShaderComponent$10,
  Godrays: ShaderComponent$$,
  Grayscale: ShaderComponent$_,
  Grid: ShaderComponent$Z,
  GridDistortion: ShaderComponent$Y,
  Group: ShaderComponent$X,
  Halftone: ShaderComponent$W,
  HexGrid: ShaderComponent$V,
  HueShift: ShaderComponent$U,
  ImageTexture: ShaderComponent$T,
  Invert: ShaderComponent$S,
  Kaleidoscope: ShaderComponent$R,
  LensFlare: ShaderComponent$Q,
  LinearBlur: ShaderComponent$P,
  LinearGradient: ShaderComponent$O,
  Liquify: ShaderComponent$N,
  Mirror: ShaderComponent$M,
  MultiPointGradient: ShaderComponent$L,
  Neon: ShaderComponent$K,
  Paper: ShaderComponent$J,
  Perspective: ShaderComponent$I,
  Pixelate: ShaderComponent$H,
  Plasma: ShaderComponent$G,
  PolarCoordinates: ShaderComponent$F,
  Polygon: ShaderComponent$E,
  Posterize: ShaderComponent$D,
  ProgressiveBlur: ShaderComponent$C,
  RadialGradient: ShaderComponent$B,
  RectangularCoordinates: ShaderComponent$A,
  Ring: ShaderComponent$z,
  Ripples: ShaderComponent$y,
  RoundedRect: ShaderComponent$x,
  Saturation: ShaderComponent$w,
  Sharpness: ShaderComponent$v,
  Shatter: ShaderComponent$u,
  SimplexNoise: ShaderComponent$t,
  SineWave: ShaderComponent$s,
  Smoke: ShaderComponent$r,
  SmokeFill: ShaderComponent$q,
  Solarize: ShaderComponent$p,
  SolidColor: ShaderComponent$o,
  Spherize: ShaderComponent$n,
  Spiral: ShaderComponent$m,
  Star: ShaderComponent$l,
  Strands: ShaderComponent$k,
  Stretch: ShaderComponent$j,
  Stripes: ShaderComponent$i,
  StudioBackground: ShaderComponent$h,
  Swirl: ShaderComponent$g,
  TiltShift: ShaderComponent$f,
  Tint: ShaderComponent$e,
  Trapezoid: ShaderComponent$d,
  Tritone: ShaderComponent$c,
  Truchet: ShaderComponent$b,
  Twirl: ShaderComponent$a,
  VHS: ShaderComponent$9,
  Vesica: ShaderComponent$8,
  Vibrance: ShaderComponent$7,
  VideoTexture: ShaderComponent$6,
  Vignette: ShaderComponent$5,
  Voronoi: ShaderComponent$4,
  WaveDistortion: ShaderComponent$3,
  Weave: ShaderComponent$2,
  WebcamTexture: ShaderComponent$1,
  ZoomBlur: ShaderComponent
};
const DEFAULT_KEY = "shaders-preview-key";
function xorCrypt(data, key) {
  const keyBytes = new TextEncoder().encode(key);
  const result = new Uint8Array(data.length);
  for (let i = 0; i < data.length; i++) {
    result[i] = data[i] ^ keyBytes[i % keyBytes.length];
  }
  return result;
}
function decodePreviewDefinition(encoded, key) {
  const binaryString = atob(encoded);
  const encrypted = new Uint8Array(binaryString.length);
  for (let i = 0; i < binaryString.length; i++) {
    encrypted[i] = binaryString.charCodeAt(i);
  }
  const decrypted = xorCrypt(encrypted, key);
  const json = new TextDecoder().decode(decrypted);
  try {
    return JSON.parse(json);
  } catch {
    throw new Error("Failed to decode shader definition: invalid data or incorrect key");
  }
}
function RenderComponent(props) {
  const Component = componentMap[props.config.type];
  if (!Component) return null;
  const componentProps = props.config.props ? {
    ...props.config.props
  } : {};
  return createComponent(Dynamic, mergeProps({
    component: Component,
    get id() {
      return props.config.id;
    },
    get renderOrder() {
      return props.config.renderOrder;
    }
  }, componentProps, {
    get children() {
      return createComponent(For, {
        get each() {
          return props.config.children;
        },
        children: (child, index) => createComponent(RenderComponent, {
          get config() {
            return {
              ...child,
              renderOrder: child.renderOrder ?? index()
            };
          },
          get structureVersion() {
            return props.structureVersion;
          }
        })
      });
    }
  }));
}
function parseStyleString(style2) {
  const map = style2.split(";").reduce((acc, declaration) => {
    const [prop, ...rest] = declaration.split(":");
    if (!(prop == null ? void 0 : prop.trim())) return acc;
    const value = rest.join(":").trim();
    const trimmed = prop.trim();
    const finalProp = trimmed.startsWith("--") ? trimmed : trimmed.replace(/-([a-z])/g, (_, c) => c.toUpperCase());
    acc[finalProp] = value;
    return acc;
  }, {});
  return map;
}
let hasWarnedLicense = false;
function Preview(props) {
  if (!hasWarnedLicense) {
    hasWarnedLicense = true;
    console.warn("[Shaders] The Preview component requires a Shaders license for production use. Visit https://shaders.com for more information.");
  }
  const [fetchedDefinition, setFetchedDefinition] = createSignal(null);
  function fetchAndDecode(url) {
    const key = props.obfuscationKey || DEFAULT_KEY;
    setFetchedDefinition(null);
    fetch(url).then((res) => {
      if (!res.ok) throw new Error(`Failed to fetch: ${res.status}`);
      return res.json();
    }).then((data) => {
      const item = data.preset || data.shader;
      if ((item == null ? void 0 : item.definition) && typeof item.definition === "string") {
        setFetchedDefinition(decodePreviewDefinition(item.definition, key));
      }
    }).catch((err) => {
      console.error("[Shaders Preview] Failed to fetch preview:", err);
    });
  }
  createEffect(() => {
    const token = props.shader;
    const baseUrl = props.apiBaseUrl || "https://shaders.com";
    if (token && typeof token === "string") {
      fetchAndDecode(`${baseUrl}/api/preview/shader/${token}`);
    } else if (!props.presetId) {
      setFetchedDefinition(null);
    }
  });
  createEffect(() => {
    const id = props.presetId;
    const baseUrl = props.apiBaseUrl || "https://shaders.com";
    if (id) {
      fetchAndDecode(`${baseUrl}/api/preview/preset/${id}`);
    } else if (!props.shader) {
      setFetchedDefinition(null);
    }
  });
  const definition = createMemo(() => {
    return fetchedDefinition() ?? null;
  });
  const watermarkText = () => props.watermarkText || "Unlock your Shaders Pro license";
  const watermarkLink = () => props.watermarkLink || "https://shaders.com/dashboard?pricing=true";
  return (() => {
    var _el$ = _tmpl$(), _el$2 = _el$.firstChild;
    insert(_el$, (() => {
      var _c$ = memo(() => !!definition());
      return () => _c$() && createComponent(Shader, {
        get colorSpace() {
          return definition().colorSpace || "p3-linear";
        },
        isPreview: true,
        style: {
          width: "100%",
          height: "100%"
        },
        get children() {
          return createComponent(For, {
            get each() {
              return definition().components;
            },
            children: (config, index) => createComponent(RenderComponent, {
              get config() {
                return {
                  ...config,
                  renderOrder: config.renderOrder ?? index()
                };
              },
              get structureVersion() {
                return definition().structureVersion;
              }
            })
          });
        }
      });
    })(), _el$2);
    _el$2.addEventListener("mouseleave", (e) => {
      e.currentTarget.style.color = "rgba(255, 255, 255, 0.5)";
    });
    _el$2.addEventListener("mouseenter", (e) => {
      e.currentTarget.style.color = "rgba(255, 255, 255, 0.8)";
    });
    insert(_el$2, watermarkText);
    effect((_p$) => {
      var _v$ = props.class, _v$2 = {
        ...typeof props.style === "string" ? parseStyleString(props.style) : props.style
      }, _v$3 = watermarkLink();
      _v$ !== _p$.e && className(_el$, _p$.e = _v$);
      _p$.t = style(_el$, _v$2, _p$.t);
      _v$3 !== _p$.a && setAttribute(_el$2, "href", _p$.a = _v$3);
      return _p$;
    }, {
      e: void 0,
      t: void 0,
      a: void 0
    });
    return _el$;
  })();
}
export {
  Preview as default
};
